File: src/main/kotlin/com/huawei/cangjie1/psi/CjTypeParameterListOwner.java
package com.huawei.cangjie1.psi;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;


public interface CjTypeParameterListOwner extends CjNamedDeclaration {
    @Nullable
    CjTypeParameterList getTypeParameterList();

    @Nullable
    CjTypeConstraintList getTypeConstraintList();

    @NotNull
    List<CjTypeConstraint> getTypeConstraints();

    @NotNull
    List<CjTypeParameter> getTypeParameters();
}

File: src/main/kotlin/com/huawei/cangjie/psi/CjTypeParameterListOwner.kt
package com.huawei.cangjie.psi

interface CjTypeParameterListOwner : CjNamedDeclaration {
    val typeParameterList: CjTypeParameterList?

    val typeConstraintList: CjTypeConstraintList?




    val typeConstraints: List<CjTypeConstraint >


    val typeParameters: List<CjTypeParameter >

}
